### ISSUE TITLE
Azure unity export - "table already exists"

### ISSUE THREAD

**Issue by offirc2** [treeverse/lakeFS#8292]
### What happened?

For Unity Catalog Export functionality in Azure Databricks, create external table in Unity Catalog fails first time with "table already exists" error message but it succeeds in 2nd attempt.

![C836CB0E-2D2C-4255-ABFB-64D46E6D587A (3)](https://github.com/user-attachments/assets/c8088fec-0472-464b-8339-138435900550)
![DEA69998-8B5F-4CFA-AF0A-71CF96E212DC (2)](https://github.com/user-attachments/assets/bb92d2db-6864-4399-a2d1-f9a2de812832)

It seems to basically the first table, the first time it tries the command, it says it failed because the table already exists. But then when we look at the command text, it runs it a second time and then it succeeds the second time around. The other thing is obviously, if we have a bunch of tables being updated at the same time and it generates a whole bunch of actions doing the same thing and some of them end up colliding and overlapping and they all fail obviously.
 
Bottom line -  it’s functional, but its just generating a lot of errors and exceptions and failures, things like that get logged kind of mostly on the lakeFS side, i guess in the action logs, but also on the sql warehouse, we can see obviously all those failed commands.

**Comment by nadavsteindler** [comment]
@offirc2 is it possible for them to share their lua script associated with this bug?

**Comment by nadavsteindler** [comment]
I met with @Jonathan-Rosenberg on this

1. Suspicion is race condition in RegisterExternalTable in client.go-- if two post-merge hooks run around the same time, they interfere with each other-- create table, delete table, etc.
2. Can we verify this by reproducing the bug on https://treeverse.us-east-1.lakefscloud.io/repositories/jonathan-unity/objects
3. How can we create a synchronization mechanism to synchronize these

**Comment by offirc2** [comment]
Customer call recording
https://us-6915.app.gong.io/call?id=3724681585591748499

@nadavsteindler @talSofer Please advise on next steps

**Comment by nadavsteindler** [comment]
OK so after the customer call, we understand that there is no requirement for concurrently exporting the same table-- this is a bit of an artificial requirement, stemming from the way the script was written(modify one table => export ALL tables)

So we are looking into a solution that allows the script to only modify the table that actually changed

**Comment by nadavsteindler** [comment]
> Customer call recording https://us-6915.app.gong.io/call?id=3724681585591748499
> 
> [@nadavsteindler](https://github.com/nadavsteindler) [@talSofer](https://github.com/talSofer) Please advise on next steps

Fix approach is here: https://docs.google.com/document/d/1F_-s64GRwDdE-2hWW_kJ_HCAuRmd-yfd6ylAnnD1cUU/edit?usp=sharing
